install_config/upgrading/blue_green_deployments.adoc

```diff
@@ -0,0 +1,166 @@
+`subscription-manager clean` to prevent the environment from being out of
+compliance.
+====
+endif::openshift-enterprise[]
+
+. xref:../../install_config/upgrading/manual_upgrades.adoc#upgrading-masters[After
+the master and etcd servers have been upgraded], you must ensure that your
+current production nodes are labeled either blue or green. In this example, the
+current installation will be blue and the new environment will be green. On each
+production node in your current installation:
++
+----
+$ oc label --all nodes color=blue
+----
++
+In the case of nodes requiring the uptime guarantees of a blue-green deployment,
+the `-l` flag can be used to match a subset of the environment using a selector.
+
+. Create the new green environment for any nodes that are to be replaced by
+xref:../install/advanced_install.adoc#adding-nodes-advanced[adding an equal
+number of new nodes] to the existing cluster. Ansible can apply the
+`*color=green*` label using the `*openshift_node_labels*` variable for each
+node.
+
+. In order to delay workload scheduling until the nodes are
+xref:../../architecture/infrastructure_components/kubernetes_infrastructure.adoc#node[healthy],
+be sure to set the `*openshift_schedulable=false*` variable.
+
+. After the green nodes are in *Ready* state, they can be made schedulable.
+
+. Blue nodes are disabled so that no new pods are run on them:
++
+----
+# oadm manage-node --schedulable=true --selector=color=green
+# oadm manage-node --schedulable=false --selector=color=blue
+----
+
+A common practice is to scale the registry and router pods until they are
+migrated to the green nodes. For these pods, a _canary_ deployment approach is
+commonly used. Scaling them up will make them immediately active on the new
+nodes. Pointing the deployment configuration to the new image initiates a
+rolling update. However, because of node anti-affinity, and the fact that the
+blue nodes are still unschedulable, the deployments to the old nodes will fail.
+At this point, the registry and router deployments can be scaled down to the
+original number of pods. At any given point, the original number of pods is
+still available so no capacity is lost.
+
+[[blue-green-deployments-warming-the-new-nodes]]
+== Warming the New Nodes
+
+In order for pods to be migrated from the blue environment to the green, the
+images must be pulled. Network latency and load on the registry can cause delays
+if there is not sufficient capacity built in to the environment. Often, the best
+way to minimize impact to the running system is to trigger new pod deployments that
+will land on the new nodes. Accomplish this by importing new image streams.
+
+A major release of {product-title} is the motivation for a blue-green
+deployment. At that time, new image streams become available for users of
+Source-to-Image (S2I). Upon import, any builds or deployments configured with
+*ImageChangeTriggers* are automatically created.
```

Comment:
@brenton Should they be instructed at this point to refer to these topics for steps?

https://docs.openshift.com/enterprise/latest/install_config/upgrading/manual_upgrades.html#updating-the-default-image-streams-and-templates
https://docs.openshift.com/enterprise/latest/install_config/upgrading/manual_upgrades.html#importing-the-latest-images